src/main/java/transact/ui/OverviewPanel.java

```diff
@@ -94,20 +99,37 @@ private void changeMonthView(YearMonth month) {
         }
     }
 
+    /**
+     * Updates the {@code monthDataMap} with information from the {@code transactionList}.
+     */
     private void updateMonthData() {
-        // TODO Implement this method
         transactionList.stream().forEach((Transaction t) -> {
-            // TODO Do Calculations to populate monthDataMap
+            if (t.getTransactionType() == TransactionType.EXPENSE) {
+                // Increase Expense
+                monthDataMap.get(YearMonth.from(t.getDate().getDate().toInstant()
```

Comment:
Getting a null pointer exception here cause monthDataMap is empty.

Need to create the empty MonthData objects and place it in the monthDataMap first.